### PR TITLE
Implement AsLL ping responses

### LINKED ISSUE #9: modify ping/pong routine to its uplink include info for AsLL
newserv's ping/pong routine with its uplink current does not support the ping time or the info for AsLL. I think it might be nice if it did. It does not need to reply to remote AsLL requests, simply providing the needed info in the ping/pong routine should make the AsLL info available on its uplink. See example below.

```
xx.cc.quakenet.org AsLL for hub.xx.quakenet.org – RTT: 2ms Upstream: -28ms Downstream: 30ms
xx.cc.quakenet.org AsLL for services.cc.quakenet.org – [unknown]
xx.cc.quakenet.org AsLL for *: 2 local servers matched
```

(originally from wiebe)

### PR DESCRIPTION
**Pull request by CuleX**
This fixes #9.

Continuation of #65, which I accidentally broke by trying to retroactively `git rebase` my fixups as not to pollute the commit history.

### CODE REVIEW

**Comment by gunnarbeutner** [comment]
Btw, you can update an existing PR by force-pushing the branch.

**Comment by CuleX** [comment]
I did force push, but because I retroactively rebased (retropc pulled master into my branch on github), all commits got re-commited by "me", which broke the entire history. When I undid that by rebasing properly on master with a separate, new commit, GitHub automatically closed the PR.

**Comment by CuleX** [comment]
Fixed a logic fail. I previously used `sms` in the branch taken when `sms == NULL`.

File: irc/irc.c [hunk 1]
```diff
@@ -611,9 +612,30 @@ void setnettime(time_t newtime) {
...
+  ss = cargv[2];
+  if ((sms = strchr(cargv[2], '.')) == NULL) {
+    diff = 0;
+  } else {
+    s = strtoull(ss, NULL, 10);
+    ms = strtoull(sms + 1, NULL, 10);
+    diff = (now.tv_sec - s) * 1000 + (now.tv_usec - ms) / 1000;
   }
+
+  irc_send("%s Z %s %s %s %llu %llu.%llu",mynumeric->content,mynumeric->content,(const char *)sender,cargv[2],diff,(unsigned long long)now.tv_sec,(unsigned long long)now.tv_usec/1000);
```
Comment (retropc) [inline irc/irc.c]: should the second arg be mynumeric->content?

only asking as it isn't on line 623!

Comment (CuleX) [reply irc/irc.c]: Line 623 was kept from the current PING/PONG mechanism.

* vararg0 (`mynumeric->content`) is the source of Z, then comes Z, then comes vararg1 (`mynumeric->content`).
* snircd [sends `sendcmdto_one(&me, CMD_PONG, sptr, "%C %s %s %i %s", &me, origin, parv[3], diff, militime_float(NULL))`](https://github.com/quakenet/snircd/blob/master/ircd/m_ping.c#L251). Source is `&me`, then comes Z, then comes its vararg0 (`&me` for %C, i.e., the numnick, see `ircd/ircd_snprintf.c` for the formta conversion).

So yes, it's supposed to be that.

Technically, the second arg [is ignored by `ms_pong` for AsLL PONGs](https://github.com/quakenet/snircd/blob/master/ircd/m_pong.c#L139-L143), but is [meant to signify the PING's destination](https://github.com/quakenet/snircd/blob/master/ircd/m_pong.c#L121), but I suppose snircd sends a the arguments this way to stay backwards compatible.

Comment (retropc) [reply irc/irc.c]: cheers

**Pull request merged**